### ISSUE TITLE
Filename of the image in header?

### ISSUE THREAD

**Issue by FritsBolt** [AdamMoses-GitHub/MMM-ImageSlideshow#12]

**Comment by FritsBolt** [comment]
How to see the filenameof the projected image in the header?

**Comment by OneOfTheInfiniteMonkeys** [comment]
Not sure if this is what you needed however I wanted the option to show the image name.
See the fork - https://github.com/OneOfTheInfiniteMonkeys/MMM-ImageSlideshow
In point of fact it's only necessary backup your 'MMM-ImageSlideshow.js' file and replace it with the forked version, should you wish to try it. 

I have not had time to test out the parameters from config.js yet. So if the parameters don't function from the config.js file. Tweak the defaults set at the top of the MMM-ImageSlideshow.js file under the section:
Module.register("MMM-ImageSlideshow", {
	// Default module config.
	defaults: { ....

Files are easily broken so take care editing the file - make backups!

Have fun.

**Comment by chassain** [comment]
Hello,

the file name in the header doesn't work...what's the problem ? thanks

**Comment by OneOfTheInfiniteMonkeys** [comment]
This thread refers to a request for the feature in this distribution of MMM-ImageSlideshow. To my knowledge, its not been pulled into this distribution at this time.

You can try, at your own risk, the distribution I forked. It's very much a development distribution, so if your not confident about backups and repairing, you might prefer to stick with this variant and the lack of image file name.

**Comment by chassain** [comment]
I have tried the distribution but no way to have name..thanks for response..
Have a good day

**Comment by OneOfTheInfiniteMonkeys** [comment]
Apologies Chassain,
Can you help me understand which distribution your tried?
Is it this one?
https://github.com/AdamMoses-GitHub/MMM-ImageSlideshow.git

**Comment by chassain** [comment]
in first yes but no position and after your git clone to have good parameters of image postion 
		{
			module: 'MMM-ImageSlideshow',
			position: 'right',
			config: {
				randomizeImageOrder: false,
				fixedImageWidth: 210,
				fixedImageHeight: 297,
				imgTitleTextPos: 1,
				imgTitleTextStyle: 'bright small light',
				slideshowSpeed: 8000,
				PathStyleText: 'fullpath',
				imagePaths: ['modules/MMM-ImageSlideshow/exampleImages'],
			}
		},

is it correct or i have to work with anoter distribution?

**Comment by OneOfTheInfiniteMonkeys** [comment]
Hi Chassain
I can't immediately see an issue, you might try the setup I have:

		{
			module: "MMM-ImageSlideshow",
			position: "middle_center",
			config: {
                                pathStyleText: "nameonly",           // display the file name only
	 			slideshowSpeed: 20000,
				imageStyleString: "object-fit: contain; width: 800px; height: 900px;",
                                imagePaths: ['modules/MMM-ImageSlideshow/exampleImages/'],
			}
		},

To see if there is a difference. Watch out if this forum pulls the tabs as the indent can be important.

**Comment by chassain** [comment]
it"s perfect..the difference is just about imageStyleString

**Comment by chassain** [comment]
the name is a little on the image with my configuration

**Comment by OneOfTheInfiniteMonkeys** [comment]
Excellent. Thanks for percevering with me.
Hope you enjoy the module, you can work with the format now and customise as you see fit.

Send me a PM on how you get on.

**Comment by chassain** [comment]
and i have the second image on the first :)....not so perfect but it's a good way

**Comment by chassain** [comment]
there is just a problem about generator because i see a image with the name and a second image behinde..

**Comment by OneOfTheInfiniteMonkeys** [comment]
No sure why you have a second image. That's not a behaviour I have ever observed. 

Possibly you have another module running that's displaying in that region.

**Comment by chassain** [comment]
yes you're right..i have 30 modules on magicmirror...Your module is good now...Do you thing it will be to delete '.jpg' for the name??? Nice job :)

**Comment by OneOfTheInfiniteMonkeys** [comment]
It can be done.
I will take a look over the weekend and try to post an update.

**Comment by chassain** [comment]
perfect..thanks..

See you ;)

**Comment by OneOfTheInfiniteMonkeys** [comment]
Have pushed an update with added features to PathStyleText parameter in config.js i.e. (RASPI /home/pi/MagicMirror/config/config.js).
Options now supported are:
'none', 'nameonly', 'fullname', 'fullpath', 'index', 'indexonly' or 'indexandtotal'

Essentially 'nameonly' should now correctly only show the image name without file extension. 

All based on the assumption a fullstop in the file name is a delimiter for the image type extension. No checking of the image binary is made.

**Comment by chassain** [comment]
Nice job man...only the name without jpg (but "." here at the end...maybe change this but it's perfect)...See you

**Comment by OneOfTheInfiniteMonkeys** [comment]
Check out the update.
Cheers.

**Comment by chassain** [comment]
you're a king :)...Perfect again

**Comment by chassain** [comment]
Hello,

Me again :)..why is it not possible to select a folder from website ? the path must be in local..
See you
